Rectangle/AppDelegate.swift

```diff
@@ -555,17 +555,34 @@ extension AppDelegate {
             func getUrlName(_ name: String) -> String {
                 return name.map { $0.isUppercase ? "-" + $0.lowercased() : String($0) }.joined()
             }
+            func getAppBundleId(_ components: URLComponents) -> String? {
+                let appBundleId = (components.queryItems?.first { $0.name == "app-bundle-id" })?.value
+                return if appBundleId == nil {
```

Comment:
Thanks for making this change here. Sorry to beleaguer this, but I figure we might as well get this tidied up. 

* What we have now can also be rewritten as `return appBundleId ?? ApplicationToggle.frontAppId` (hence one of the reasons why you don't typically see `if` come after `return`)
* However, this would still be the same as `return appBundleId` since once we get to `ApplicationToggle`, we would be providing the `ApplicationToggle.frontAppId` at that time in the case of `appBundleId` being nil.
* If we want to account for `app-bundle-id=` in the URL, then we would have to check that `appBundleId != nil && appBundleId.isEmpty` (depending on if we do indeed get back an empty string in that case - I didn't test it).

I'm still ok either way on doing the empty string check, but it still makes the most sense to me to at the very least change this to `return appBundleId` here.

Reply:
Sorry, I pushed a change in the train yesterday as a reflex, (if anything happen to the laptop).

I agree with your points; my original goal is indeed the third point.